buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/PrecommitTasks.groovy

```diff
@@ -96,9 +103,62 @@ class PrecommitTasks {
         }
         Task forbiddenApis = project.tasks.findByName('forbiddenApis')
         forbiddenApis.group = "" // clear group, so this does not show up under verification tasks
+
         return forbiddenApis
     }
 
+    private static Task configureForbiddenApisCli(Project project) {
+        project.configurations.create("forbiddenApisCliJar")
+        project.dependencies {
+            forbiddenApisCliJar 'de.thetaphi:forbiddenapis:2.5'
+        }
+        Task forbiddenApisCli = project.tasks.create('forbiddenApisCli')
+        project.ext.getForbiddenSignaturesFile = { name -> project.getForbiddenSignaturesFile(name) }
+
+        project.sourceSets.forEach { sourceSet ->
+            forbiddenApisCli.dependsOn(
+                project.tasks.create(sourceSet.getTaskName('forbiddenApisCli', null), JavaExec) {
+                    classpath = project.files(
+                            project.configurations.forbiddenApisCliJar,
+                            sourceSet.compileClasspath,
+                            sourceSet.runtimeClasspath
+                    )
+                    main = 'de.thetaphi.forbiddenapis.cli.CliMain'
+                    executable = "${project.runtimeJavaHome}/bin/java"
+                    [
+                            'jdk-unsafe-1.8', 'jdk-deprecated-1.8', 'jdk-non-portable', 'jdk-system-out'
+                    ].forEach { args "-b", it }
+                    [
+                            getForbiddenSignaturesFile(project, 'jdk-signatures'),
+                            getForbiddenSignaturesFile(project, 'es-all-signatures')
+                    ].forEach { args "-f", it }
+                    if (sourceSet.name == 'test') {
+                        args "-f", getForbiddenSignaturesFile(project, 'es-test-signatures')
+                        args "-f", getForbiddenSignaturesFile(project, 'http-signatures')
+                    } else {
+                        args "-f", getForbiddenSignaturesFile(project, 'es-server-signatures')
+                    }
+                    args "--suppressannotation", '**.SuppressForbidden'
+                    dependsOn sourceSet.classesTaskName
+                    doFirst {
+                        // Forbidden APIs expects only existing dirs, and requires at least one
+                        FileCollection existingOutputs = sourceSet.output.classesDirs
+                                .filter { it.exists() }
+                        if (existingOutputs.isEmpty()) {
+                            throw new StopExecutionException("${sourceSet.name} has no outputs")
+                        }
+                        existingOutputs.forEach { args "-d", it }
+                    }
+                }
+            )
+        }
+        return forbiddenApisCli
+    }
+
+    private static File getForbiddenSignaturesFile(project, String name) {
```

Comment:
Could you add a type to the `project` parameter?

Reply:
I haven't found a good way to fight IDEA on this ... 
The refactoring don't add a type. Will change it.